scheme/arm-cca/evidence_handler.go

```diff
@@ -117,84 +115,48 @@ func (s EvidenceHandler) AppraiseEvidence(
 	ec *proto.EvidenceContext, endorsementsStrings []string,
 ) (*ear.AttestationResult, error) {
 	var endorsements []handler.Endorsement // nolint:prealloc
-
-	result := handler.CreateAttestationResult(SchemeName)
-
-	for i, e := range endorsementsStrings {
-		var endorsement handler.Endorsement
-
-		if err := json.Unmarshal([]byte(e), &endorsement); err != nil {
-			return nil, fmt.Errorf("could not decode endorsement at index %d: %w", i, err)
-		}
-
-		endorsements = append(endorsements, endorsement)
-	}
-
-	err := populateAttestationResult(result, ec.Evidence.AsMap(), endorsements)
-
-	// TO DO: Handle Unprocessed evidence when new Attestation Result interface
-	// is ready. Please see issue #105
-	return result, err
-}
-
-func populateAttestationResult(
-	result *ear.AttestationResult,
-	evidence map[string]interface{},
-	endorsements []handler.Endorsement,
-) error {
-	claims, err := common.MapToClaims(evidence["platform"].(map[string]interface{}))
+	var err error
+	var status ear.TrustTier
+	endorsements, err = getEndorsementsFromString(endorsementsStrings)
 	if err != nil {
-		return err
+		return nil, err
 	}
+	var claims map[string]interface{}
+	CcaPlatform := &CCAPlatformAttester{}
+	CcaRealm := &CCARealmAttester{}
+	evidence := ec.Evidence.AsMap()
 
-	appraisal := result.Submods[SchemeName]
-
-	// once the signature on the token is verified, we can claim the HW is
-	// authentic
-	appraisal.TrustVector.Hardware = ear.GenuineHardwareClaim
+	result := handler.CreateAttestationResult("CCA_SSD_PLATFORM")
 
-	rawLifeCycle, err := claims.GetSecurityLifeCycle()
+	/* Perform SubAttester Appraisal */
+	claims = evidence["platform"].(map[string]interface{})
+	appraisal := result.Submods["CCA_SSD_PLATFORM"]
+	err = CcaPlatform.PerformAppraisal(appraisal, claims, endorsements)
 	if err != nil {
-		return handler.BadEvidence(err)
+		return nil, err
 	}
-
-	lifeCycle := psatoken.CcaLifeCycleToState(rawLifeCycle)
-	if lifeCycle == psatoken.CcaStateSecured ||
-		lifeCycle == psatoken.CcaStateNonCcaPlatformDebug {
-		appraisal.TrustVector.InstanceIdentity = ear.TrustworthyInstanceClaim
-		appraisal.TrustVector.RuntimeOpaque = ear.ApprovedRuntimeClaim
-		appraisal.TrustVector.StorageOpaque = ear.HwKeysEncryptedSecretsClaim
-	} else {
-		appraisal.TrustVector.InstanceIdentity = ear.UntrustworthyInstanceClaim
-		appraisal.TrustVector.RuntimeOpaque = ear.VisibleMemoryRuntimeClaim
-		appraisal.TrustVector.StorageOpaque = ear.UnencryptedSecretsClaim
+	claims = evidence["realm"].(map[string]interface{})
+	realmappraisal := &ear.Appraisal{
+		TrustVector: &ear.TrustVector{},
+		Status:      &status,
 	}
```

Comment:
Not sure why you need to do this, wouldn't it be sufficient to create a default object?

```suggestion
	realmappraisal := &ear.Appraisal{}
```

(While you are here, maybe use camel-case for the variable name: `realmAppraisal`)

Reply:
> Not sure why you need to do this, wouldn't it be sufficient to create a default object?
> 
> (While you are here, maybe use camel-case for the variable name: `realmAppraisal`)

No this is required otherwise just creating the way you suggested would trap reason `nil` TrustVector

Reply:
I will change to camel-case instead ..

Reply:
> > Not sure why you need to do this, wouldn't it be sufficient to create a default object?
> > (While you are here, maybe use camel-case for the variable name: `realmAppraisal`)
> 
> No this is required otherwise just creating the way you suggested would trap reason `nil` TrustVector

Sorry, I don't get the logic.  Is `PerformAppraisal` using `realmAppraisal` as an in-out parameter?  If not - as I'd guess - just create the object in it and return it, no need to pass it as a parameter, no?

Reply:
@thomas-fossati : The suggested change makes the PerformAppraisal() methid look different between, `CCA Platform` & `CCA Realm` which I wanted to avoid, hence I passed the appraisal structure pointer as first argument, which will be modified by the `subAttesters`

Reply:
I understand the "uniformity of interfaces" argument, but that doesn't prevent modifying both :-)

The point is that there is no reason for the caller pre-allocate the object, it's entirely the responsibility of the `PerformAppraisal` functions to do that.